### ISSUE TITLE
Blank page in PDF view (Windows)

### ISSUE THREAD

**Issue by 19joho66** [frescobaldi/frescobaldi#1222]
I've build a new poppler.dll and python-poppler-qt5.pyd. When using these libraries, the resulting pdf page is blank. 
The score is rendered, because point and click works (clicking in source code jumps to the place in the pdf view and frames the position and the other way around). It almost feels like the score is beeing printed in the background color.
Any ideas what could cause this issue? Does the linux version display the score poperly?

**Comment by uliska** [comment]
No idea, just confirming that it works properly on Linux.
Mac anyone?

**Comment by wbsoft** [comment]
Which versions of poppler and python-poppler-qt5 are you using?

**Comment by 19joho66** [comment]
Poppler-0.82 and python-poppler-qt5-0.75

**Comment by wbsoft** [comment]
Can you try a shell session with python3 in Windows? And then in Python (by heart:-):
```python
from PyQt5.Qt import *
a = QApplication([])
import popplerqt5
doc = popplerqt5.Poppler.Document.load("music.pdf")
img = doc.page(0).renderToImage()
img.size()
img.save("test.png")
<Ctrl+D> to exit Python
```
And then view the resulting image.
(music.pdf is a pdf generated by LilyPond)

**Comment by 19joho66** [comment]
It's the same result - a blank page.

**Comment by wbsoft** [comment]
Can you attach the pdf you are using?

**Comment by 19joho66** [comment]
I've build poppler version 0.68 and that works fine, so it seems that poppler has a problem. 

@wbsoft: Would you still like to have a look at the pdf and image of the not working version?

**Comment by wbsoft** [comment]
Yes, please! And maybe you can do some tests within Frescobaldi whether the Arthur backend of poppler 0.68 is usable. And a big thank you for all your help! :-)

**Comment by 19joho66** [comment]
[document.pdf](https://github.com/frescobaldi/frescobaldi/files/3895765/document.pdf)
![test](https://user-images.githubusercontent.com/15361970/69705199-5823e200-10f5-11ea-8a37-b9be8e156a6c.png)

What do I have to do to test the Arthur backend?

**Comment by 19joho66** [comment]
This is the image of the working version (poppler 0.68):

(https://user-images.githubusercontent.com/15361970/69705399-c5d00e00-10f5-11ea-8981-85825176b224.png)

**Comment by 19joho66** [comment]
I've done some more builds:
poppler-0.80 works
poppler-0.81 and poppler-0.82 do not

**Comment by wbsoft** [comment]
Thank you! (With size I meant the output of `img.size()` in python (the width and length) :-) because I thought the image might have had 0x0 size. But that's not the case; clearly the bug is in poppler)

If you have current Frescobaldi from a git checkout, you can test the Arthur backend by enabling "Use Vector-based backend (Arthur)  on screen" (and/or printing) in Preferences→Music Preferences→Display of Music.

I found that only recent Poppler versions have usable Arthur support. Thanks to Arthur, we can print vector graphics (which is better than raster images) from within Frescobaldi without cups, and also export pieces of a PDF to vector-based SVG images.

**Comment by wbsoft** [comment]
Maybe you can report an issue with Poppler: https://gitlab.freedesktop.org/poppler/poppler/issues/
See also https://poppler.freedesktop.org/

**Comment by 19joho66** [comment]
This is what it looks like on the Screen:
No Arthur
![No Arthur ](https://user-images.githubusercontent.com/15361970/69729687-1318a380-1127-11ea-98d9-58612abd71f3.png)

With Arthur
![With Arthur ](https://user-images.githubusercontent.com/15361970/69729696-17dd5780-1127-11ea-81bf-22cbcdb387a7.png)

**Comment by wbsoft** [comment]
Is this poppler 0.80 ?

**Comment by 19joho66** [comment]
Yes, it is, but this version is build with absolutely minimal dependencies (just freefont). Maybe something else is also needed for Arthur (like cairo, libjpeg ....)?

**Comment by 19joho66** [comment]
I've just compiled the lates poppler-0.83 which is working again.
But still no go on Arthur. This is the error message I get in the command line window when I activate Arthur:

```
QWindowsFontDatabase::fontEngine: Can't change family name of font
RawFont is not valid
QWindowsFontDatabase::fontEngine: Can't change family name of font
RawFont is not valid
RawFont is not valid
QWindowsFontDatabase::fontEngine: Can't change family name of font
RawFont is not valid
RawFont is not valid
RawFont is not valid
RawFont is not valid
RawFont is not valid
QWindowsFontDatabase::fontEngine: Can't change family name of font
RawFont is not valid
QWindowsFontDatabase::fontEngine: Can't change family name of font
RawFont is not valid
RawFont is not valid
RawFont is not valid
RawFont is not valid
RawFont is not valid
RawFont is not valid
RawFont is not valid

```
Maybe that helps...

**Comment by wbsoft** [comment]
Maybe you can report this to the poppler developers. It would be nice if we could get this to work on Windows. I have no windows machine to help with debugging.

**Comment by 19joho66** [comment]
I've reported this problem at poppler an it seems that under windows there is a problem with free fonts. When using the poppler-qt5viewer.exe (a demo program from the poppler developers) this problem occurs also with lilypond pdfs as well as LaTex documents. 
I've found a solution for this program by adding a qt.config files whicht contains 
I've found a very simple solution: Add a qt.conf file in the folder containing the exe file which contains
```
[Platforms]
WindowsArguments = fontengine=freetype
```
Then everything is displayed all right with arthur (At least with the demo program). The file tells windows to use the freefont engine.

Now I need to find out how to include the qt.conf file as a resource under PyQt5 and then everything hopefully works under windows and arthur...

**Comment by wbsoft** [comment]
Thank you for your great debugging work!

**Comment by rualark** [comment]
Hi! Could you please tell where poppler-qt5 binaries for Windows 10 can be downloaded, which works with frescobaldi?

**Comment by 19joho66** [comment]
I've build the poppler-qt5.dll myself. There are some other dependent 
dlls needed too. I can send them to you tomorrow if you like.
I still haven't figured out how to get arthur working in the pyqt5 
environment, but at least it can be used for the standard pdf view.

Am 2019-12-17 22:15, schrieb Alexey Arkhipenko:

**Comment by rualark** [comment]
Thanks a lot. Could you please zip needed files and attach it to comment (link below comment) so that everyone could download it if needed. Like this:
[python35dll.zip](https://github.com/frescobaldi/frescobaldi/files/3975890/python35dll.zip)

**Comment by 19joho66** [comment]
[python_poppler_qt5-0.83.0-cp35-cp35m-win32.zip](https://github.com/frescobaldi/frescobaldi/files/3978006/python_poppler_qt5-0.83.0-cp35-cp35m-win32.zip)

Install the whl file via pip.  Please report if this works for you.

**Comment by rualark** [comment]
Thanks. Could you also publish dll file so that a windows user without developer skills can use Frescobaldi (without using pip and knowing what whl is)? Where does user need to copy dll file for Frescobaldi to show PDF correctly?

**Comment by 19joho66** [comment]
You could try to place the three files from the poppler.zip archive into the frescobaldi folder. That should work (hopefully)

[poppler.zip](https://github.com/frescobaldi/frescobaldi/files/3982120/poppler.zip)

**Comment by rualark** [comment]
Thanks for your help! I extracted these files to `C:\Program files (x86)\Frescobaldi` but I see the same message when I start Frescobaldi: `Could not load the popplerqt5 module`. Where can I see more detailed error message? For example, if it did not find some file or file was wrong.

**Comment by 19joho66** [comment]
I've found a way to fix the Arthur problem in Windows. I'v made a PR (#1246) for this because I would like to have the change tested in the Linux and Mac Version before merging it. 
Could someone please evaluate this?

**Comment by 19joho66** [comment]
Seems to be fixed with Version 3.1.1
